### PR TITLE
Add check for packaging information

### LINKED ISSUE #22: Check if pyproject.toml defines library correctly
Would be nice to have a check for whether the library should be labelled as a module or package and then verify that is the case.  Would prevent faulty installations, since these aren't currently detected during the build process.

### PR DESCRIPTION
**Pull request by tekktrik**
Implements fix created by https://github.com/adafruit/actions-ci-circuitpython-libs/pull/19

### CODE REVIEW

**Comment by tekktrik** [comment]
@FoamyGuy this is the preventative fix for the issue you encountered!

File: build/action.yml [hunk 1]
```diff
@@ -59,16 +59,11 @@ runs:
...
-  - name: Install dependencies
-    shell: bash
-    run: |
-      pip install -r requirements.txt
-      pip install -r optional_requirements.txt
   - name: Run pytest
     shell: bash
     run: |
       if [[ -d tests ]]; then
-        pip install .
```
Comment (jepler) [inline build/action.yml:72]: would it make sense to move the install of pytest down to here as well?

**Review by jepler — COMMENTED** [review]
while looking at #23 I realized that this is related to an issue I saw while building documentation: for RTD, I think that `docs/requirements.txt` is used. However, this CI uses the top-level `requirements.txt` only.

Here's the command [used by RTD](https://readthedocs.org/projects/jepler-udecimal/builds/19929636/):
```
python -m pip install --exists-action=w --no-cache-dir -r docs/requirements.txt 
```

For the purposes of jepler_udecimal I have listed sphinx-autoapi in optional-requirements.txt as well as docs/requirements.txt, but #23 will switch the "optional" deps to install only when pytest is used. This won't affect jepler_udecimal since it does happen to contain pytest tests.

In particular, this means that essentially all of the docs built during CI will be broken, because sphinxcontrib-jquery won't be installed...

another alternative is to drop doc building from github actions and widely enable this instead: [Preview documentation from pull requests](https://docs.readthedocs.io/en/stable/pull-requests.html#preview-documentation-from-pull-requests)

File: build/action.yml (hunk 1 above)
Comment (tekktrik) [reply build/action.yml:72]: Ah, it would!

**Comment by tekktrik** [comment]
I think the documentation preview is interesting but may take some time to implement since I think every library, as it's own project, would need this enabled, and make sure it's enabled for new projects too.  @kattni, what are your thoughts?

I think the intention for "optional dependencies" was for either opt-in behavior (e.g., using PIL on CPython/Blinka libraries) or things used in examples (e.g., something like a Python library for interacting with Discord).  In this case, I think it's worth having an unconditional install of `docs/requirements.txt` added and keep the installation of truly "optional" dependencies separate.  Does that sound like a good solution?  I agree have docs built correctly is important, even in the build CI.

**Comment by tekktrik** [comment]
@jepler is this all set for review?  Please let me know if I haven't addressed anything from your last comment!

**Comment by jepler** [comment]
I think I need to explain again the problem I see:

In practice, the `optional_requirements.txt` file is used to install dependencies used during the CI documentation-building step that occurs on github (even though the result of that is never used). I noticed this in jepler_udecimal but this is also seen in nunchuk within the standard bundle according to a search. I didn't check the community bundle.

But after the CI changes here, `optional_requirements.txt` is only used *if there are tests*, which means that a library that doesn't have tests, but has docs with dependencies in `optional_requirements.txt` will fail to build, I *think*.

**Comment by tekktrik** [comment]
I actually noticed (and fixed) in #29 the fact that the pip installations here were entirely redundant, and confirmed it by checking the CI.  They actually get installed beforehand via https://github.com/adafruit/actions-ci-circuitpython-libs/blob/820e76313197a5721a966d7bd3cfc70ad8156a61/install.sh#L23-L26 so that line would also be redundant now.  I believe I understand what you're saying now, but I think this means nothing should break for now.

Relatedly, it may be necessary to patch the libraries to add `optional_requirements.txt` to `.readthedocs.yaml` then, based on what you're saying.

**Comment by tekktrik** [comment]
Changes made, I can bring up the patch idea this Monday at the weekly meeting!  Pinging @kattni in advance in case we need to go that route.

**Review by jepler — APPROVED** [review]
thanks for clearing that up, I appreciate it.

**Comment by tekktrik** [comment]
Thanks for helping me out with this!

**Pull request merged**